nats.go

```diff
@@ -1803,6 +1803,21 @@ func (nc *Conn) ConnectedUrl() string {
 	return nc.current.url.String()
 }
 
+// ConnectedUrlRedacted reports the connected server's URL with passwords redacted
```

Comment:
if you connect with like `nats://user:password@foo:4222` `ConnectedUrl()` returns the password in clear, this just an easier way to get something safely logable